apps/mailman/mailer.py

```diff
@@ -74,12 +84,16 @@ def send(self):
         text_content = self.text_template.render(c)
         html_content = self.html
 
-        msg = EmailMultiAlternatives(self.subject, text_content,
-                                     self.from_email, self.to_email)
-        msg.content_subtype = "html"
-        msg.attach_alternative(html_content, "text/html")
+        connection = get_connection()
+        messages = []
+        for recipient in self.to_email:
+            message = EmailMultiAlternatives(self.subject, text_content,
+                                             self.from_email, [recipient])
+            message.content_subtype = "html"
+            message.attach_alternative(html_content, "text/html")
+            messages.append(message)
         try:
-            msg.send()
+            connection.send_messages(messages)
```

Comment:
This sends each message individually. This is different than bcc, but it keeps the to: line to a single address, and was easier to test locally without a mail server.

Reply:
Hi Annie,

-Both logos are located here: https://www.fairelectionscenter.org/, but the work elections one is a bit small. Would uploading that one on a hidden page on that site work to get the right size logo online?

-The Survey Monkey URLs aren't dummy, but I did use the same one over and over. The URLs are currently auto-populated using our survey email creator but have been set up on the back end by @devseedgit 
We would like the template email to be receptive to having it work the same way. Selecting a jurisdiction and having the survey link populate on the email, but instead of the link on its own, have a button populate, if possible. 

-I'm not sure I understand the local mail server question. In the past with different email platforms I've been able to send a test in plain text and html. Checking it on my email and phone for formatting and other issues. I realize that different email servers will look differently (Outlook vs Gmail), I'm not sure what the response to this should be.

-Would sending each message individually make us have to send messages one by one? We wanted to have the option to do a bulk email selecting the different recipients all at once so we can save time in not having to make them one by one.

Reply:
> -Both logos are located here: https://www.fairelectionscenter.org/, but the work elections one is a bit small. Would uploading that one on a hidden page on that site work to get the right size logo online

I think it would work, but I'm not super familiar with writing these types of mails.  I worry that if there are a lot (thousands) of recipients, it can stress the server the images are hosted on. I assumed that's why the original email had an image on a CDN (https://s3.amazonaws.com/voteworker/workelections.png). I think it would be best to put both images there, but I don't know how they got there. @devseedgit do you know?

> -The Survey Monkey URLs aren't dummy, but I did use the same one over and over. The URLs are currently auto-populated using our survey email creator but have been set up on the back end by @devseedgit
We would like the template email to be receptive to having it work the same way. Selecting a jurisdiction and having the survey link populate on the email, but instead of the link on its own, have a button populate, if possible.

Sorry for the confusion here! I'm still using the same code to generate the survey URLs, just with buttons instead of links. All the buttons in the screenshot were auto-generated from the list of jurisdictions I selected in the UI. But on a local server, the code produces URLs like http://example.com734, because the local server is configured to use 'http://example.com'. Just wanted to check that it's configured correctly in production, I guess @devseedgit is the one to ask?

> -I'm not sure I understand the local mail server question. In the past with different email platforms I've been able to send a test in plain text and html. Checking it on my email and phone for formatting and other issues. I realize that different email servers will look differently (Outlook vs Gmail), I'm not sure what the response to this should be.

@devseedgit any best practices for testing the emails generated for correct CSS in Gmail and Outlook? Right now I've got raw mail printed to the console, and no local mailserver.

> -Would sending each message individually make us have to send messages one by one? We wanted to have the option to do a bulk email selecting the different recipients all at once so we can save time in not having to make them one by one.

This still takes the full list of emails it did before, but now if you list 3 emails, it sends 3 emails, each with one of the addresses on the to: line. Before, it sent one email with 3 addresses on the to: line.

Reply:
Ok, it sounds like we're on the page here. I'll standby for what @devseedgit @alyssadelaine @scisco say regarding q's address to them

-CND image upload on the amazon location
-local email server

Reply:
@anniesullie I think you should be able to use the django [backend mail server](https://docs.djangoproject.com/en/2.1/topics/email/) to test sending the e-mails, and the SurveyMonkey URL is configured correctly in production, and you can test it locally by setting it as a local environment variable; in that case, it will overwrite `example.com`